src/services/LibraryService.ts

```diff
@@ -89,4 +97,36 @@ export class LibraryService implements Service<fhir4.Library> {
 
     return createLibraryPackageBundle(library[0], req.query['include-terminology'] ?? false);
   }
+
+  /**
+   * result of sending a POST request to:
+   * {BASE_URL}/4_0_1/Library/$submit or {BASE_URL}/4_0_1/Library/:id/$submit
+   * POSTs a new artifact in "draft" status. The operation results in an error if the artifact
+   * does not have status set to "draft."
+   */
+  async submit(args: RequestArgs, { req }: RequestCtx) {
+    logger.info(`${req.method} ${req.path}`);
+
+    const contentType: string | undefined = req.headers['content-type'];
+    checkContentTypeHeader(contentType);
+
+    const resource = req.body;
+    checkExpectedResourceType(resource.resourceType, 'Library');
+
+    // check for "draft" status on the resource
+    if (resource.status !== 'draft') {
+      throw new BadRequestError(`The artifact must be in 'draft' status.`);
+    }
+
+    // lastUpdated should be second because it should overwrite a meta.lastUpdated tag in the request body
```

Comment:
I am a little confused about this comment...looks like it was leftover from the implementation of create in the test-server. Maybe clarify or just take it out?

Reply:
It's basically saying that when we update `resource.meta` we should take the existing contents of `resource.meta` and then overwrite the `lastUpdatedTag` that currently exists on `resource.meta`.

However, I just looked the [spec](http://www.hl7.org/fhir/resource-definitions.html#Resource.meta) and saw this: "This value [of `lastUpdated`] is always populated except when the resource is first being created." So I think this can be removed entirely (and will probably need to be removed from deqm test server at some point because it appears in the `baseCreate` fn when it should only be in `baseUpdate`)